apis/v1beta1/httproute_types.go

```diff
@@ -244,6 +244,56 @@ type HTTPRouteRule struct {
 	// +optional
 	// +kubebuilder:validation:MaxItems=16
 	BackendRefs []HTTPBackendRef `json:"backendRefs,omitempty"`
+
+	// Timeouts defines the timeouts that can be configured for an HTTP request.
+	//
+	// Support: Extended
+	//
+	// +optional
+	// <gateway:experimental>
+	Timeouts *HTTPRouteTimeouts `json:"timeouts,omitempty"`
+}
+
+// HTTPRouteTimeouts defines timeouts that can be configured for an HTTPRoute.
+// Timeout values are formatted like 1h/1m/1s/1ms as parsed by Golang time.ParseDuration
+// and MUST BE >= 1ms or 0 to disable (no timeout).
```

Comment:
@robscott

I think the upper bound is limited by the internal representation of `time.Duration` which is `int64`. So that's ~300 years? So the limit is controlled by `time.ParseDuration`. Not sure if it truncates or fails to parse string values that won't fit? But I was assuming that documenting this value is according to `time.ParseDuration` is all we really need to say. I suppose we could make up some smaller value and add validation for that, but I'm not sure that's worth it.

Also, if IIUC, the Rust implementation of duration is represented by an unsigned int, so the max value might be higher. Bottom line is I don't think this value really needs to be documented in the spec because the bottom line is that the max is huge, and nobody will ever use such huge values anyway.

The only exception is "no timeout", hang forever until the client kills the connection. Envoy does in fact use 0 for "no timeout". This was something I added to address reviewer feedback of the GEP from @candita.

Reply:
I created #2155 to follow up on this - at a high level, I think we really need to start with the tightest possible validation here and only loosen if/when it's needed. Having clearly defined limits enables implementations to test against min and max values to ensure that they can support the full range of possible inputs from the API.

Reply:
For reference, Linkerd also interprets 0 as "no timeout": https://github.com/linkerd/linkerd2/blob/main/policy-controller/k8s/index/src/outbound/index.rs#L341